collector/transport_node_collector.go

```diff
@@ -81,8 +81,30 @@ func (c *transportNodeCollector) generateTransportNodeStatusMetrics() (transport
 		} else {
 			status = 0
 		}
-		transportNodeStatusMetric := prometheus.MustNewConstMetric(c.transportNodeStatus, prometheus.GaugeValue, status, transportNode.Id, transportNode.DisplayName)
+		transportZoneLabels := c.buildTransportZoneLabels(transportNode)
+		desc := c.buildStatusDesc(transportZoneLabels)
+		transportNodeStatusMetric := prometheus.MustNewConstMetric(desc, prometheus.GaugeValue, status, transportNode.Id, transportNode.DisplayName)
 		transportNodeMetrics = append(transportNodeMetrics, transportNodeStatusMetric)
 	}
 	return
 }
+
+func (c *transportNodeCollector) buildStatusDesc(extraLabels map[string]string) *prometheus.Desc {
+	return prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "transport_node", "status"),
+		"Status of Transport Node UP/DOWN",
+		[]string{"id", "name"},
+		extraLabels,
+	)
+}
+
+func (c *transportNodeCollector) buildTransportZoneLabels(transportNode manager.TransportNode) map[string]string {
```

Comment:
seems we don't use other attributes than `TransportZoneEndpoints` can we just pass this intead of the whole transportNode?

Reply:
with this, we need to rename the method to buildTransportZoneEndpointLabels, doing otherwise will create confusion between the method name and parameter

Reply:
sure